tests/test_trace.py

```diff
@@ -171,5 +172,40 @@ def test_two_level_flattened_hf_bert():
     assert not find_module_in_graph(sub_sch.mod.graph, "attention.self.query")
 
 
+def test_dynamo():
+    from transformers import AutoConfig, BertLMHeadModel
+
+    config = AutoConfig.from_pretrained("bert-large-uncased")
+    with slapo.init_empty_weights():
+        model = BertLMHeadModel(config)
+    sch = slapo.create_schedule(model)
+    subsch = sch[f"bert.encoder.layer.{0}.attention.self"]
+
+    bs, seq_length = 8, 512
+    concrete_args = {"hidden_states": torch.randn(bs, seq_length, config.hidden_size)}
+    sig = inspect.signature(subsch.mod.forward)
+    concrete_args.update(
+        {
+            p.name: p.default
+            for p in sig.parameters.values()
+            if p.name not in concrete_args
+        }
+    )
+    subsch.trace(tracer="dynamo", concrete_args=concrete_args)
+    assert isinstance(subsch.mod, fx.GraphModule)
+
+    def pattern(x):
+        x = call_module(r"self_(query|key|value)", x)
+        x = x.view((8, 512, 16, 64))
+        return x.permute(0, 2, 1, 3)
+
+    qkv_subgraphs = subsch.find(pattern)
```

Comment:
The find API supports function as input, do we have a design doc for this? 
It is not intuitive to me, the `pattern` function seems to directly modify the computation semantics.

Reply:
See #35. We only *find* the pattern but have not changed the computation graph here. This is used to find the Q/K/V dense layer and the consequential elementwise operators, shown in
https://github.com/huggingface/transformers/blob/main/src/transformers/models/gpt_neo/modeling_gpt_neo.py#L168-L174